components/script/dom/webgl2renderingcontext.rs

```diff
@@ -0,0 +1,908 @@
+    fn BlendColor(&self, r: f32, g: f32, b: f32, a: f32) {
+        self.base.BlendColor(r, g, b, a)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
+    fn BlendEquation(&self, mode: u32) {
+        self.base.BlendEquation(mode)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
+    fn BlendEquationSeparate(&self, mode_rgb: u32, mode_alpha: u32) {
+        self.base.BlendEquationSeparate(mode_rgb, mode_alpha)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
+    fn BlendFunc(&self, src_factor: u32, dest_factor: u32) {
+        self.base.BlendFunc(src_factor, dest_factor)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
+    fn BlendFuncSeparate(&self, src_rgb: u32, dest_rgb: u32, src_alpha: u32, dest_alpha: u32) {
+        self.base.BlendFuncSeparate(src_rgb, dest_rgb, src_alpha, dest_alpha)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.9
+    fn AttachShader(&self, program: Option<&WebGLProgram>, shader: Option<&WebGLShader>) {
+        self.base.AttachShader(program, shader)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.9
+    fn DetachShader(&self, program: Option<&WebGLProgram>, shader: Option<&WebGLShader>) {
+        self.base.DetachShader(program, shader)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.9
+    fn BindAttribLocation(&self, program: Option<&WebGLProgram>,
+                          index: u32, name: DOMString) {
+        self.base.BindAttribLocation(program, index, name)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
+    fn BindBuffer(&self, target: u32, buffer: Option<&WebGLBuffer>) {
+        self.base.BindBuffer(target, buffer)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.6
+    fn BindFramebuffer(&self, target: u32, framebuffer: Option<&WebGLFramebuffer>) {
+        self.base.BindFramebuffer(target, framebuffer)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.7
+    fn BindRenderbuffer(&self, target: u32, renderbuffer: Option<&WebGLRenderbuffer>) {
+        self.base.BindRenderbuffer(target, renderbuffer)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.8
+    fn BindTexture(&self, target: u32, texture: Option<&WebGLTexture>) {
+        self.base.BindTexture(target, texture)
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.8
+    fn GenerateMipmap(&self, target: u32) {
+        self.base.GenerateMipmap(target)
+    }
+
+    #[allow(unsafe_code)]
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
+    unsafe fn BufferData(&self, cx: *mut JSContext, target: u32, data: *mut JSObject, usage: u32) -> Fallible<()> {
+        self.base.BufferData(cx, target, data, usage)
```

Comment:
Ugh, it'd be nice to forward all these a bit more easily :(.

Can you file an issue to maybe generate macros from `CodegenBinding.py` or something like that? Not sure if it's common enough to be worth it, but...

Reply:
I didn't find this pattern in another API. WebGL 2.0 WebIDL interfaces instead of inheritance.

At least I generated all this stuff with a local script ;) I'll open that issue to see if it's worth for more APIs